### PR TITLE
menu item crud apis

### PR DESCRIPTION
**Pull request by asifmons**

### CODE REVIEW

File: src/main/java/com/vendor/caterer/dto/MenuItemCreateRequest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import lombok.Data;
+import org.springframework.data.elasticsearch.annotations.Document;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+@Data
+public class MenuItemCreateRequest {
```
Comment (sijosaji) [inline src/main/java/com/vendor/caterer/dto/MenuItemCreateRequest.java:14]: CatererId?

File: src/main/java/com/vendor/caterer/dto/MenuItemUpdateRequest.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+public class MenuItemUpdateRequest {
+    private String name;
+    private String description;
+    private BigDecimal price;
+    private boolean isVeg;
+    private boolean isDrink;
+    private String imageUrl;
+    private List<String> tagsToAdd;
+    private List<String> tagsToRemove;
+    private Set<UUID> categoryIdsToAdd;
```
Comment (sijosaji) [inline src/main/java/com/vendor/caterer/dto/MenuItemUpdateRequest.java:19]: Can it be updated to a model for  categoryIdsToAdd  like 
CateroryUpdate{
Operation type; // Add/Delete;
Set<UUID> ids;
}

File: src/main/java/com/vendor/caterer/helper/CollectionUtils.java
Comment (sijosaji) [inline src/main/java/com/vendor/caterer/helper/CollectionUtils.java:1]: Why you need this class there is a inbuilt class for this

File: src/main/java/com/vendor/caterer/services/MenuItemService.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            MenuItem updatedMenuItem = convertUpdateRequestModelToDocModel(menuItem.get(), updateRequest);
+            dao.save(updatedMenuItem);
+            return ResponseEntity.status(HttpStatus.OK).body(updatedMenuItem);
+        } else {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+        }
+    }
+
+    private MenuItem convertUpdateRequestModelToDocModel(MenuItem menuItem, MenuItemUpdateRequest updateRequest) {
+        //MenuItem updatedMenuItem = MenuItemMapper.mapper.mapUpdateRequestToModel(updateRequest, menuItem);
```
Comment (sijosaji) [inline src/main/java/com/vendor/caterer/services/MenuItemService.java:55]: Why are you not using mapper?

**Review by sijosaji — COMMENTED** [review]
Pls go through the comments.

File: src/main/java/com/vendor/caterer/dto/MenuItemCreateRequest.java (hunk 1 above)
Comment (asifmons) [reply src/main/java/com/vendor/caterer/dto/MenuItemCreateRequest.java:14]: done

File: src/main/java/com/vendor/caterer/helper/CollectionUtils.java
Comment (asifmons) [reply src/main/java/com/vendor/caterer/helper/CollectionUtils.java:1]: ok this class is removed.

File: src/main/java/com/vendor/caterer/dto/MenuItemUpdateRequest.java (hunk 2 above)
Comment (asifmons) [reply src/main/java/com/vendor/caterer/dto/MenuItemUpdateRequest.java:19]: done

File: src/main/java/com/vendor/caterer/services/MenuItemService.java (hunk 3 above)
Comment (asifmons) [reply src/main/java/com/vendor/caterer/services/MenuItemService.java:55]: mapper is now used.

**Comment by asifmons** [comment]
changes based on first stage review comments are completed.

**Review by sijosaji — APPROVED** [review]
Looks Good

**Pull request merged**